### DISCUSSION TITLE
Experimental to native

### DISCUSSION THREAD

**Discussion by PilarJ** [mletenay/home-assistant-goodwe-inverter#173] [General]
Hello, 

will this new features be transformed into native HA integration? I am now using native HA and need this new features for automatization, so I am asking if I have to change integration into this experimental or it will be added into native in some days.

Thanks for answer

**Comment by mletenay** [comment]
Some of these "experimental" features may get into core HA after they are matured and tested enough (e.g. Eco Charge mode).
But some (like the services) may not get there at all or at least not any time soon.
So if you need them just install HACS and this "experimental" version, it is really drop-in replacement and you can go back to core HA anytime without loosing any data.
I intend to maintain both core and HACS version alongside as long as necessary.

**Reply by PilarJ** [reply]
ok, thanks for answer
